src/plugins/data_source/server/legacy/configure_legacy_client.ts

```diff
@@ -194,14 +194,14 @@ const getBasicAuthClient = async (
 };
 
 const getAWSClient = (credential: SigV4Content, clientOptions: ConfigOptions): LegacyClient => {
-  const { accessKey, secretKey, region, service } = credential;
+  const { accessKey, secretKey, region } = credential;
   const client = new LegacyClient({
     connectionClass: HttpAmazonESConnector,
     awsConfig: new Config({
       region,
       credentials: new Credentials({ accessKeyId: accessKey, secretAccessKey: secretKey }),
     }),
-    service,
+
```

Comment:
I believe it is not needed. The only usage of `getAWSClient` in the repo is line 120. const `awsClient` is used in `addClientToPool` and `callAPI.bind(null, awsClient)` which `getQueryClient` then returns. None of them should use service to my knowledge/investigation, and also type `legacyClient` / class `Client` does not have this service property so that should be a good indication.

https://github.com/opensearch-project/OpenSearch-Dashboards/blob/bca1dcfcc783ccf2837c26f41bd6c5305aa9f869/src/plugins/data_source/server/legacy/configure_legacy_client.ts#L120

Reply:
Thanks for the clarification!

Reply:
I did notice, however, that there is a test expecting `service` to still be included. 
I have yet to figure out why this might be needed. still researching.

https://github.com/CMDWillYang/OpenSearch-Dashboards/blob/1c0f44422994a9e5da13a4807dc5812a35b20026/src/plugins/data_source/server/legacy/configure_legacy_client.test.ts#L167-L179